From 81fb2268cf9aadde4a995c92f571073d3843857d Mon Sep 17 00:00:00 2001
From: Christophe Kerello <christophe.kerello@foss.st.com>
Date: Tue, 15 Jun 2021 12:07:59 +0200
Subject: [PATCH 124/841] feat(mtd): spi_nor: use spi_mem_dirmap_read

Use spi_mem_dirmap_read instead of spi_mem_exec_op when reading data
from the memory.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 666d2b7d6379fd99647ec4508690fb70fe4fc7f1 feat(mtd): spi_nor: use spi_mem_dirmap_read]
Change-Id: Icd4c6974c3f6fee11c04a660f6c31f1b8230a51c
Signed-off-by: Christophe Kerello <christophe.kerello@foss.st.com>
---
 drivers/mtd/nor/spi_nor.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mtd/nor/spi_nor.c b/drivers/mtd/nor/spi_nor.c
index 2e343448a..477e5e5b7 100644
--- a/drivers/mtd/nor/spi_nor.c
+++ b/drivers/mtd/nor/spi_nor.c
@@ -300,7 +300,7 @@ int spi_nor_read(unsigned int offset, uintptr_t buffer, size_t length,
 			nor_dev.read_op.data.nbytes = length;
 		}
 
-		ret = spi_mem_exec_op(&nor_dev.read_op);
+		ret = spi_mem_dirmap_read(&nor_dev.read_op);
 		if (ret != 0) {
 			spi_nor_clean_bar();
 			return ret;
-- 
2.47.2

